### PR TITLE
first commit adding_data_augmentation

### PR DESCRIPTION
**Pull request by johanobandoc**
First commit of the scrip adding_data_augmentation. What we do here is include all the components necessary for using the following function:

`functools.partial(atari_100k_rainbow_agent.preprocess_inputs_with_augmentation,             data_augmentation=data_augmentation)`

### CODE REVIEW

File: lifting_veil/agents/dqn_agent_new.py [hunk 1]
```diff
@@ -244,6 +247,18 @@ def __init__(self,
...
         self._rng = jax.random.PRNGKey(seed)
+
+
+        if preprocess_fn == "data_augmentation":
+            self._preprocess_fn = functools.partial(atari_100k_rainbow_agent.preprocess_inputs_with_augmentation,
+                data_augmentation=data_augmentation)
+        elif preprocess_fn is None:
+            self.network_def = network(num_actions=num_actions)
+            self._preprocess_fn = networks.identity_preprocess_fn
+        else:
```
Comment (psc-g) [inline lifting_veil/agents/dqn_agent_new.py:258]: Do you need this condition? When do you expect to use it?

File: lifting_veil/agents/dqn_agent_new.py [hunk 2]
```diff
@@ -264,8 +279,10 @@ def __init__(self,
                                                        initzer=self._initzer,
                                                        normalization=self._normalization,
                                                        layer_funct=self._layer_funct),
-                             optimizer=optimizer,
+                             otimizer=optimizer,
```
Comment (psc-g) [inline lifting_veil/agents/dqn_agent_new.py]: typo?

Comment (joaogui1) [inline lifting_veil/agents/dqn_agent_new.py]: why this change? Is it a typo?

File: lifting_veil/agents/dqn_agent_new.py [hunk 3]
```diff
@@ -299,6 +316,9 @@ def _train_step(self):
             if self.training_steps % self.update_period == 0:
                 self._sample_from_replay_buffer()
 
+                states = self._preprocess_fn(self.replay_elements['state'])
```
Comment (psc-g) [inline lifting_veil/agents/dqn_agent_new.py]: you need to pass an rng here:
https://github.com/google/dopamine/blob/master/dopamine/labs/atari_100k/atari_100k_rainbow_agent.py#L104

File: lifting_veil/agents/dqn_agent_new.py [hunk 4]
```diff
@@ -388,7 +408,8 @@ def begin_episode(self, observation):
         if not self.eval_mode:
             self._train_step()
 
-        self._rng, self.action = select_action(self.network_def, self.online_params, self.state, self._rng,
+        state = self._preprocess_fn(self.state)
+        self._rng, self.action = select_action(self.network_def, self.online_params, state, self._rng,
```
Comment (psc-g) [inline lifting_veil/agents/dqn_agent_new.py:413]: you also need to modify the step function with the preprocessed state

**Review by psc-g — CHANGES_REQUESTED** [review]
